examples/ecs-fargate/prisma/infra/index.js

```diff
@@ -2,37 +2,93 @@ const { PrismaClient } = require('@prisma/client');
 
 const pkg = require('./package.json');
 
+function getPrefix() {
+  const lowerCaseLetters = Array(26)
+    .fill('a')
+    .map((c, i) => String.fromCharCode(c.charCodeAt() + i));
+  const digits = Array(10)
+    .fill('0')
+    .map((c, i) => String.fromCharCode(c.charCodeAt() + i));
+  const chars = [lowerCaseLetters, digits].flat();
+  const randChar = ()  => chars[Math.floor(Math.random() * chars.length)];
+  const randLetter = () => lowerCaseLetters[Math.floor(Math.random() * lowerCaseLetters.length)];
+  return (
+    randLetter() +
+    Array(6)
+      .fill('')
+      .map(() => randChar())
+      .join('')
+  );
+}
+
 // TODO replace with your desired project name
 const appName = pkg.name;
+const pipelineName = `${appName}-pipeline`;
+const prefix = getPrefix();
 const cbRole = `${appName}codebuild`;
-const ghToken = process.env.GH_PAT;
+const cpRole = `${appName}codepipeline`;
 const region = process.env.AWS_REGION;
 const port = 8088;
+const bucketName = `${prefix}-${appName}-bucket`;
+
+
+const codepipelinePolicyArn = 'arn:aws:iam::aws:policy/AWSCodePipelineFullAccess';
 const codebuildPolicyArn = 'arn:aws:iam::aws:policy/AWSCodeBuildAdminAccess';
 const cloudwatchLogsArn = 'arn:aws:iam::aws:policy/CloudWatchLogsFullAccess';
 const pushEcrPolicyArn = 'arn:aws:iam::aws:policy/EC2InstanceProfileForImageBuilderECRContainerBuilds';
+const s3PolicyArn = 'arn:aws:iam::aws:policy/AmazonS3FullAccess';
+
 const assumeServicePolicy = {
-  "Statement": [
+  Statement: [
     {
-      "Effect": "Allow",
-      "Principal": {
-        "Service": "codebuild.amazonaws.com"
+      Effect: 'Allow',
+      Principal: {
+        Service: 'codepipeline.amazonaws.com',
       },
-      "Action": "sts:AssumeRole"
+      Action: 'sts:AssumeRole',
     },
   ],
-  "Version": "2012-10-17"
+  Version: '2012-10-17',
+};
+const assumeServicePolicyCodebuild = {
+  Statement: [
+    {
+      Effect: 'Allow',
+      Principal: {
+        Service: 'codebuild.amazonaws.com',
+      },
+      Action: 'sts:AssumeRole',
+    },
+  ],
+  Version: '2012-10-17',
+};
+
+const cbRoleData = {
+  role_name: cbRole,
+  assume_role_policy_document: assumeServicePolicyCodebuild,
+  attached_policies_arns: [codebuildPolicyArn, cloudwatchLogsArn, pushEcrPolicyArn, s3PolicyArn]
+}
+const cpRoleData = {
+  role_name: cpRole,
+  assume_role_policy_document: assumeServicePolicy,
+  attached_policies_arns: [codepipelinePolicyArn, codebuildPolicyArn, s3PolicyArn],
 };
-const ghUrl = 'https://github.com/iasql/iasql-engine';
 
-const prisma = new PrismaClient()
+const cpBucketData = {
+  name: bucketName,
+  region: region
+}
+
+const ghUrl = 'iasql-engine';
```

Comment:
if the url is not needed anymore maybe this variable should be called `ghRepositoryName` or something like that?